.github/workflows/ousd.yml

```diff
@@ -0,0 +1,147 @@
+          cache-dependency-path: contracts/yarn.lock
+
+      - run: yarn install --frozen-lockfile
+        working-directory: ./contracts
+
+      - run: yarn run lint
+        working-directory: ./contracts
+
+      - run: yarn prettier:check
+        working-directory: ./contracts
+
+  contracts-test:
+    name: "Contracts Unit Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: contracts/yarn.lock
+
+      - run: yarn install --frozen-lockfile
+        working-directory: ./contracts
+
+      - run: yarn run test
+        working-directory: ./contracts
+
+  contracts-forktest:
+    name: "Contracts Fork Tests"
+    runs-on: ubuntu-latest
+    env:
+      HARDHAT_CACHE_DIR: ./cache
+      PROVIDER_URL: ${{ secrets.PROVIDER_URL }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: contracts/yarn.lock
+
+      - uses: actions/cache@v3
+        id: hardhat-cache
+        with:
+          path: contracts/cache
+          key: ${{ runner.os }}-hardhat-${{ hashFiles('contracts/cache/*.json') }}
+          restore-keys: |
+            ${{ runner.os }}-hardhat-cache
+
+      - run: yarn install --frozen-lockfile
+        working-directory: ./contracts
+
+      - run: yarn run test:fork
+        working-directory: ./contracts
+
+  dapp-lint:
+    name: "DApp Linter"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: dapp/yarn.lock
+
+      - run: yarn install --frozen-lockfile
+        working-directory: ./dapp
+
+      - run: yarn prettier:check
+        working-directory: ./dapp
+
+  slither:
+    name: "Slither"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 3.6
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+
+      - name: Install dependencies
+        run: |
+          wget https://github.com/ethereum/solidity/releases/download/v0.5.11/solc-static-linux
+          chmod +x solc-static-linux
+          sudo mv solc-static-linux /usr/local/bin/solc
+          pip3 install slither-analyzer
+
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: contracts/yarn.lock
+
+      - run: yarn install --frozen-lockfile
+        working-directory: ./contracts
+
+      - name: Test with Slither
+        working-directory: ./contracts
+        run: yarn run slither
+
+  snyk:
```

Comment:
Is the Snyk integration still working? we stopped paying for the subscription a while ago because our team did not find it useful.
Perhaps we could get rid of it and enable GitHub dependabot as a replacement? Those two tools are slightly different but if we keep up with updating dependencies recommended by dependabot I think that will go a long way in terms of keeping the repo safe against package related security vulnerability. Thoughts?

Reply:
I do see it working [on the workflow](https://github.com/OriginProtocol/origin-dollar/actions/runs/3097657322/jobs/5014583055). Looks like their free plan works with OpenSource projects, there's a limit though.

I'll look into adding dependabot as a separate PR, will leave this until we have that done

Reply:
Here's [an issue](https://github.com/OriginProtocol/origin-dollar/issues/1115) to keep track of it